cmake/external/mkldnn.cmake

```diff
@@ -77,3 +77,16 @@ FILE(WRITE ${dummyfile} "const char * dummy = \"${dummyfile}\";")
 ADD_LIBRARY(mkldnn STATIC ${dummyfile})
 TARGET_LINK_LIBRARIES(mkldnn ${MKLDNN_LIB} ${MKLML_LIB} ${MKLML_IOMP_LIB})
 ADD_DEPENDENCIES(mkldnn ${MKLDNN_PROJECT})
+
+# copy the real so.0 lib to install dir
+# it can be directly contained in wheel or capi
+SET(MKLDNN_SHARED_LIB ${MKLDNN_INSTALL_DIR}/libmkldnn.so.0)
+ADD_CUSTOM_COMMAND(OUTPUT ${MKLDNN_SHARED_LIB}
+    COMMAND cp ${MKLDNN_LIB} ${MKLDNN_SHARED_LIB}
+    DEPENDS mkldnn)
+ADD_CUSTOM_TARGET(mkldnn_shared_lib ALL DEPENDS ${MKLDNN_SHARED_LIB})
+
+IF(WITH_C_API)
+  INSTALL(FILES ${MKLDNN_SHARED_LIB} DESTINATION lib)
```

Comment:
Not stall to `lib`, but `third_party/mkldnn/lib`. Please refer to https://github.com/PaddlePaddle/Paddle/blob/develop/cmake/external/glog.cmake#L71

Reply:
Thanks @Xreki , The reason installed to `lib` is that the script [here ](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/scripts/docker/build.sh#L206) installed to `lib` directly.
This function is trying to replace the command.

And I think the *.so should be contained into `/usr/local/lib`, maybe @Yancey1989 has considered this before?
Anyway I am OK installing to third-party, but then user should set the path to LD manually.

Same as below.

Reply:
As glog etc is a static library, but mkldnn etc is a shared library, thus, if *.so are contained into `thirdparty`, users should `export LD_LIBRARY_PATH` which leads an extra step for users. So we can contain *.so into `/usr/loacl/lib` at this PR.

Reply:
The same with @luotao1 , I think `*.so` contained into `/usr/local/lib` is an easy way for the user.